Program.cs

```diff
@@ -19,10 +20,12 @@ public static void Main(string[] args) {
 			builder.Logging.AddConsole();
 
 			// Get the name of the serial port where data is arriving
-			Console.WriteLine("Enter the name of the serial port where the APC220 module is connected.\nAvailable ports are:");
-			Console.Write(string.Concat(SerialPort.GetPortNames().Select(x => "\t" + x + "\n")) + "> ");
-			string serialPortName = Console.ReadLine()!;
-
+			var serialPortName = AnsiConsole.Prompt(
+				new SelectionPrompt<string>()
+					.Title("Please select the serial port where the [blue]APC220 module[/] is connected.")
+					.PageSize(10)
+					.AddChoices(SerialPort.GetPortNames().Select(x => x))
+			);
```

Comment:
The menu is erased after selecting a port - either find a way to make it persist, with the selected option highlighted, or add a simple log afterwards.

Reply:
Done.